src/core/text-buffer.cc

```diff
@@ -723,10 +726,10 @@ void TextBuffer::reset(Text &&new_base_text) {
   top_layer->previous_layer = nullptr;
 }
 
-Patch TextBuffer::get_inverted_changes(const Snapshot *snapshot) const {
-  vector<const Patch *> patches;
-  Layer *layer = top_layer;
-  while (layer != &snapshot->base_layer) {
+Patch TextBuffer::get_inverted_changes(const shared_ptr<Snapshot> snapshot) const {
+  vector<const Patch*> patches;
+  auto layer = top_layer;
+  while (layer.get() !=  &snapshot->base_layer) {
```

Comment:
Not sure about this

Reply:
Looks like, for comparison only, the gets aren't needed. But I'd recommend keeping them to be explicit that the shared pointers themselves are not being compared.